bootstrap/config/config.go

```diff
@@ -552,9 +630,76 @@ func (cp *Processor) listenForChanges(serviceConfig interfaces.Configuration, co
 	}()
 }
 
-// logConfigInfo logs the config info message with number over overrides that occurred.
-func (cp *Processor) logConfigInfo(message string, overrideCount int) {
-	cp.lc.Infof("%s (%d envVars overrides applied)", message, overrideCount)
+func (cp *Processor) waitForCommonConfig(configClient configuration.Client) error {
+	// Wait for configuration provider to be available
+	isAlive := false
+	for cp.startupTimer.HasNotElapsed() {
+		if configClient.IsAlive() {
+			isAlive = true
+			break
+		}
+
+		cp.lc.Warnf("Waiting for configuration provider to be available")
+
+		select {
+		case <-cp.ctx.Done():
+			return errors.New("aborted waiting Configuration Provider to be available")
+		default:
+			cp.startupTimer.SleepForInterval()
+			continue
+		}
+	}
+	if !isAlive {
+		return errors.New("configuration provider is not available")
+	}
+
+	// check to see if common config is loaded
+	isConfigReady := false
+	for cp.startupTimer.HasNotElapsed() {
+		commonConfigReady, err := configClient.GetConfigurationValue(config.CommonConfigDone)
+		if err != nil {
+			return fmt.Errorf("failed to retrieve common config value %s: %s", config.CommonConfigDone, err.Error())
+		}
+		isCommonConfigReady, err := strconv.ParseBool(string(commonConfigReady))
+		if err != nil {
+			return fmt.Errorf("failed to parse common config value %s for key %s: %s", commonConfigReady, config.CommonConfigDone, err.Error())
+		}
```

Comment:
Need to treat these errors as Consul up, but not ready and retry. This was the issue I fixed in core-common-config-bootstrapper